variables.tf

```diff
@@ -1,11 +1,252 @@
+}
+
+variable "max_scan_concurrency" {
+  default     = 20
+  description = "The maximum number of instances to operate on at once"
+  type        = number
+}
+
+variable "max_scan_errors" {
+  default     = 20
+  description = "The maximum number of errors before stopping the install task scheduling"
+  type        = number
+}
+
+variable "max_install_concurrency" {
+  default     = 2
+  description = "The maximum number of instances to operate on at once"
+  type        = number
+}
+
+variable "max_install_errors" {
+  default     = 2
+  description = "The maximum number of errors before stopping the install task scheduling"
+  type        = number
+}
+
+variable "platforms" {
+  description = "The list of platforms you want to support"
+  type        = set(string)
+
+  default = [
+    "AMAZON_LINUX_2",
+    "AMAZON_LINUX",
+    "CENTOS",
+    "ORACLE_LINUX",
+    "SUSE",
+    "WINDOWS",
+    "DEBIAN",
+    "UBUNTU",
+    "REDHAT_ENTERPRISE_LINUX",
+    "MACOS"
+  ]
+
+  # Make sure that only supported platforms are passed in
+  validation {
+    error_message = "Please check that you are specifying only supported platforms."
+
+    condition = (
+      toset(var.platforms) == setintersection(
+        var.platforms,
+        [
+          "AMAZON_LINUX_2",
+          "AMAZON_LINUX",
+          "CENTOS",
+          "ORACLE_LINUX",
+          "SUSE",
+          "WINDOWS",
+          "DEBIAN",
+          "UBUNTU",
+          "REDHAT_ENTERPRISE_LINUX",
+          "MACOS"
+        ]
+      )
+    )
+  }
+}
+
+variable "log_bucket" {
+  default     = null
+  description = "S3 bucket that logs should be sent to"
+  type        = string
+}
+
+variable "scan_cutoff" {
+  default     = 1
+  description = "How many hours before the end of the maintenance Window to stop scheduling new instances to scan"
+  type        = number
+}
+
+variable "scan_duration" {
+  default     = 4
+  description = "How long in hours for the scan maintenance window"
+  type        = number
+}
+
+variable "scan_log_prefix" {
+  default     = "/patch_manager/scan/"
+  description = "The S3 bucket subfolder to store scan logs in"
+  type        = string
+}
+
+variable "scan_notification_configs" {
+  default     = []
+  description = "A set of objects containing `notification_config`s [docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ssm_maintenance_window_task#notification_config)"
+
+  type = set(object({
+    notification_arn    = string
+    notification_events = string
+    notification_type   = string
+  }))
+
+  validation {
+    error_message = "Invalid value for `notification_events`."
+
+    condition = (
+      length(var.scan_notification_configs) == 0 ||
+      toset(flatten(
+        [
+          for config in var.scan_notification_configs : config.notification_events
+        ]
+        )) == setintersection(
+        [
+          "All",
+          "InProgress",
+          "Success",
+          "TimedOut",
+          "Cancelled",
```

Comment:
Did AWS misspell canceled? If they did can we get a comment with that info?

Reply:
Both spellings are correct, one `l` is just more common in American English.

But yes, this is the way AWS spelled it.